wrpc/peer.go

```diff
@@ -62,8 +66,17 @@ func (p *Peer) AddConn(conn *Conn) {
 		if err != nil {
 			p.ErrLogger(fmt.Errorf("read thread: %w", err))
 		}
-		// TODO(hbagdi): handle error
-		_ = p.conn.Close()
+		err2 := p.conn.Close()
+		if err2 != nil {
+			p.ErrLogger(fmt.Errorf("closing wrpc connection: %w", err2))
```

Comment:
If err is nil, then this error will be handled twice: logged once here and then passed onto CloseCallbackFunc as well. Is that intended?

Reply:
first thought:  is there any rule of thumb on when should a library log an error and when to just return them?
second thought: if we're returning all errors, then we don't need the `ErrLogger` callback.
third thought:  This is the only call to `ErrLogger`, we can use it to remove the Node from Koko, no need at all for this PR!

Reply:
(well, just need to make sure _some_ function is called on every close, even without error)

Reply:
The general rule is that an error should be handled once - either logged once or returned, doing both should be avoided. There are exceptions but I can't see an exception here.

Reply:
And yes, generally, libs should return errors when possible.